individual_modules/intermediate_version_control/branches_and_merging.ipynb

```diff
@@ -158,9 +158,31 @@
     "``` sh\n",
     "git merge feature-branch\n",
     "```\n",
+    "### Merge Conflicts\n",
     "\n",
-    "The merge will conclude automatically if the changes were on different lines.\n",
-    "If the changes overlapped between the two branches, we must manually resolve the conflicts.\n",
+    "A **merge conflict** occurs when Git cannot automatically combine changes from two branches. \n",
+    "This usually happens when two people edit the same line in a file or when one person edits a file that was deleted in another branch.\n",
+    "\n",
+    "Merge conflicts happen because Git needs a clear instruction on which change to keep when multiple edits affect the same part of a file.\n",
+    "To resolve a merge conflict:\n",
+    "\n",
+    "1. Identify the conflicting files (Git will show them after a merge attempt).\n",
+    "2. Open the file and look for conflict markers: \n",
+    "\n",
+    "- `<<<<<<< HEAD` → start of the conflict in your current branch  \n",
+    "- `=======` → separator between your changes and incoming changes  \n",
+    "- `>>>>>>> branch-name` → end of the conflict from the other branch\n",
+    "\n",
+    "3. Decide which changes to keep or combine the changes manually.\n",
+    "4. Remove the conflict markers and save the file.\n",
+    "5. Stage the resolved file using:\n",
+    "```sh\n",
+    "git add <file>\n",
+    "```\n",
+    "6. Complete the merge with:\n",
+    "```sh\n",
+    "git commit\n",
+    "```\n",
```

Comment:
Could add an extra short note on IDEs, something like

"Most modern code editors should help you find any merge conflicts and provide buttons/shortcuts for resolving the changes, but the above steps will work everywhere."